### ISSUE TITLE
Feat for infusing weapons with spells

### ISSUE THREAD

**Issue by itamarcu** [mlenser/kryx-rpg-issues#653]
This is a suggestion for a feat intended to evoke the flavor of a gish (usually Magus or Infuser) that enchants or infuses their weapons, often ranged weapons, with effects.

It's already been a Magus class feature before, and this feat could remain limited to bonded weapons if it's deemed necessary.  The goal is to have the feat invoke a lot of flavor, and give some limited extra versatility to your spells (allowing you to extend their range in a way).

It should solve the problems mentioned in #264 - allowing for the flavor niche of arcane archers.

It should also (at its current draft) work perfectly with the War Magic feature, allowing magi to channel cantrips into weapon attacks instead of casting them separately (this mostly matters for flavor).

# General Feat - Arcane Shot - 6th level
*Prerequisites:  the ability to cast at least one spell*
You can channel your magic into your ranged weapons, turning them into vectors of attack for your spells.  When you cast a spell with a casting time of one or two actions, you can choose to store its energy within your weapon or piece of ammunition instead of applying its effects immediately.
The next time you make an attack with that weapon before the end of your next turn, that spell is released as if you cast it from the point at which the projectile landed, with any areas of effect originating from that position.  You are considered to be touching the target of your attack, unless you missed your attack roll.
For example, you could charge your arrow to release a *[frost nova](https://www.kryxrpg.com/spells/frost-nova)* centered on a point on the ground, infuse your crossbow bolt with a [*lightning bolt*](https://www.kryxrpg.com/spells/lightning-bolt) that streaks through your enemies, or throw an dagger imbued with [*knock/lock*](https://www.kryxrpg.com/spells/knock-lock) at a door.

---

### Balance Concerns
- This feat allows for increased range with certain spells - spells with a range of 30 feet could have their range increased to 120 feet this way, or even more if the archer makes long range attacks.  This is strong, but the archer is already specializing in ranged combat, and in many situations the archer will have to make a hard choice between targeting the perfect spot or wasting the potential weapon attack damage.  The [Attach Bomb](https://www.kryxrpg.com/feats/attach-bomb) feat allows for something very similar, at the cost of disadvantage on the attack (and it could be added here too if needed).
- This feat allows you to cast line and cone spells that originate from a spot other than you.  This should be okay, as the [Directional Bomb](https://www.kryxrpg.com/feats/directional-bomb) feat already allows for that.
- This feat allows you to cast some "spheres centered on yourself" spells at a range.  This is potentially strong, but it's intentional, because it perfectly fits the flavor of e.g. shooting arrows that explode into fire.  It couldn't really be done with spells like Fireball, because they already have a long range so the archer wouldn't have any reason to cast them through a weapon (not to mention the fact it costs 2 actions, making it a higher cost in combat).
- This feat technically allows for a preemptive buff before combat starts - if a player enchants their weapon with a spell.  It requires the player to be able to cast spells silently and commit to the fight.  If it's still too useful, the "before the end of your next turn" limitation could be changed to "before the start of your next turn", but then it would prevent usage of any 2-action spells.

**Comment by mlenser** [comment]
Added Imbued Weapon as a general feat.

I changed some factors:
- The spell cannot be a sphere emanating from you: These spells are much stronger than ranged AoE spells so allowing them to be used as ranged AoE spells would make them too strong
- The spell is released when the weapon hits. This is better for melee weapons. For a projectile I made it so the projectile lands within 1.5 meters of the target if the attack misses by 4 or less.
